src/components/Timeline/PTimelineItem.vue

```diff
@@ -0,0 +1,236 @@
+<template>
+  <li class="p-timeline-item" :class="classes.root">
+    <slot>
+      <div v-if="slots.left" class="p-timeline-item__left-container" :class="classes.leftContainer">
+        <slot name="left" />
+      </div>
+
+      <div class="p-timeline-item__center-container" :class="classes.centerContainer">
+        <div class="p-timeline-item__point-container">
+          <slot name="point">
+            <PTimelinePoint />
+          </slot>
+        </div>
+      </div>
+
+      <div v-if="slots.right" class="p-timeline-item__right-container" :class="classes.rightContainer">
+        <slot name="right" />
+      </div>
+    </slot>
+  </li>
+</template>
+
+<script lang="ts" setup>
+  import { computed, useSlots } from 'vue'
+  import { TimelineAlignment, TimelineLayout } from '@/types/timeline'
+
+  const slots = useSlots()
+
+  const props = defineProps<{
+    align?: TimelineAlignment,
+    layout?: TimelineLayout,
+  }>()
+
+  const align = computed(() => props.align ?? 'left')
+  const layout = computed(() => props.layout ?? 'default')
+
+  const classes = computed(() => {
+    const stacked = layout.value === 'stacked'
+    return {
+      root: {
+        'p-timeline-item--align-left': align.value === 'left',
+        'p-timeline-item--align-right': align.value === 'right',
+        'p-timeline-item--align-center': align.value === 'center',
+        'p-timeline-item--layout-default': layout.value === 'default',
+        'p-timeline-item--layout-stacked': stacked,
+        'p-timeline-item--layout-alternate': layout.value === 'alternate',
+      },
+      centerContainer: {
+        'p-timeline-item__center-container--layout-stacked': stacked,
+        'p-timeline-item__center-container--align-right': stacked && align.value === 'right',
+        'p-timeline-item__center-container--align-center': stacked && align.value === 'center',
+        'p-timeline-item__center-container--align-left': stacked && align.value === 'left',
+      },
+      leftContainer: {
+        'p-timeline-item__left-container--layout-stacked': stacked,
+        'p-timeline-item__left-container--align-right': stacked && align.value === 'right',
+        'p-timeline-item__left-container--align-center': stacked && align.value === 'center',
+        'p-timeline-item__left-container--align-left': stacked && align.value === 'left',
+      },
+      rightContainer: {
+        'p-timeline-item__right-container--layout-stacked': stacked,
+        'p-timeline-item__right-container--align-right': stacked && align.value === 'right',
+        'p-timeline-item__right-container--align-center': stacked && align.value === 'center',
+        'p-timeline-item__right-container--align-left': stacked && align.value === 'left',
+      },
```

Comment:
I could clean these computed classes up by interpolating the align and layout values... if that's clearer when reading the code I don't mind reworking this a bit

Reply:
+1 I think that would simplify this quite a bit.